tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/AppServiceTests.cs

```diff
@@ -47,5 +53,27 @@ await mi.AcquireTokenForManagedIdentity(ManagedIdentityTests.Resource)
                 Assert.AreEqual(string.Format(CultureInfo.InvariantCulture, MsalErrorMessage.ManagedIdentityEndpointInvalidUriError, "IDENTITY_ENDPOINT", "127.0.0.1:41564/msi/token", AppService), ex.Message);
             }
         }
+
+        [DataTestMethod]
```

Comment:
nit: not sure I get all these permutations. Maybe write a regression test that focuses on this exact bug (and metion the bug)